lale/lib/aix360/beam_search.py

```diff
@@ -0,0 +1,320 @@
+import numpy as np
```

Comment:
Hi  @hirzel  currently I am using the v0.1.0 branch of aix360(https://github.com/Trusted-AI/AIX360/tree/v0.1.0)  and not the latest branch. The latest(0.2.1) branch of AIX360 has some dependency issues. It supports hard dependency using tensorflow ==1.14 and it is not available for python 3.8 through pip dependencies. It causing  Lale build failure .
0.1.0 version does not have any such hard dependency for tensorflow but it has some bugs so we monkey patched above classes  from 0.2.1 version to 0.1.0 and now the build is passing and the code is also working fine.

Reply:
Thanks for explaining, it is good to know your rationale behind copying this code. That said, we would like to avoid having code copied from other open-source packages. Given the versioning and dependency issues, I recommend creating an appropriate test matrix in [build.yml](https://github.com/IBM/lale/blob/master/.github/workflows/build.yml).

Reply:
HI @hirzel ,

Could you please   elaborate more for "creating an appropriate test matrix in build.yml"?

Thanks

Reply:
You said that AIX360 0.2.1 uses TensorFlow 1.14 and it is not available for Python 3.8. So I think you should test your contributions inside of a test matrix that installs AIX360 0.2.1 with TensorFlow 1.14 and Python 3.7. That way, your contributions can use AIX360 as-is without copy-and-pasting all that code. By the way, the TensorFlow version will probably not matter, because the functionality exercised by your code does not seem to use TensorFlow.

Reply:
Hi @hirzel  could you please guide me in creation of the mentioned test matrix.

Reply:
The goal is to have a test that uses the latest AIX360, instead of copy-and-pasting code from AIX360 to Lale. According to your earlier comments, the latest AIX360 has some dependency conflicts with Lale. So what you need is a setup in [build.yml](https://github.com/IBM/lale/blob/97a2b3f83b37c31c61819ea59178f770236765a7/.github/workflows/build.yml) that runs AIX360 tests with dependencies that work. If you look at the file, you can see how to restrict the Python version for a given test, for example, to [Python 3.7](https://github.com/IBM/lale/blob/97a2b3f83b37c31c61819ea59178f770236765a7/.github/workflows/build.yml#L86-L87). Also, you can see how to install a specific version of some other packages for a given test, such as [scikit-learn and lightgbm](https://github.com/IBM/lale/blob/97a2b3f83b37c31c61819ea59178f770236765a7/.github/workflows/build.yml#L115-L117). I'm guessing you don't need to change the scikit-learn or lightgbm versions, but you may need to play with the tensorflow version, or even try to `pip uninstall` tensorflow, since it is not needed.